src/cursor.c

```diff
@@ -52,8 +49,8 @@ static void Cursor_RemoveFromIdle(Cursor *cur) {
   }
 
   Array_Resize(idle, sizeof(Cursor *) * (n - 1));
-  if (cur->nextTimeoutNs == cur->parent->nextIdleTimeoutNs) {
-    cur->parent->nextIdleTimeoutNs = 0;
+  if (cur->nextTimeoutMs == cur->parent->nextIdleTimeoutMs) {
+    cur->parent->nextIdleTimeoutMs = 0;
```

Comment:
So you changed the cardinality from NS to MS?

Reply:
Yes. using `RedisModule_Milliseconds`
I doubt this caused the issue...

getting the expiration in the test to 2.5 may fix the test...
Do you have another idea for fix? Code itself seems solid